### ISSUE TITLE
[Clustering] num_clusters must >= 1?

### ISSUE THREAD

**Issue by cailurus** [pycaret/pycaret#1339]
Hey guys, I'm trying to do clustering for geodata, so I think KMeans should be the first try.

For now, I want to find the center of my dataset. I can do this easily with sklearn

```
kmeans = KMeans(n_clusters=1, random_state=0).fit(X)
kmeans.cluster_centers_
```

But in pycaret I got: 

```
/opt/conda/lib/python3.8/site-packages/pycaret/internal/tabular.py in create_model_unsupervised(estimator, num_clusters, fraction, ground_truth, round, fit_kwargs, verbose, system, raise_num_clusters, X_data, display, **kwargs)
   2488     # checking num_clusters type:
   2489     if num_clusters <= 1:
-> 2490         raise TypeError(
   2491             "num_clusters parameter can only take value integer value greater than 1."
   2492         )

TypeError: num_clusters parameter can only take value integer value greater than 1.

```

So, could you guys tell me a little bit more about this restriction? I mean, why we set 'num_clusters >= 1' ?

**Comment by babaniyi** [comment]
To use KMeans clustering, It is advisable to use > 1 cluster. The reason is that for KMeans to work:
1. Select K random points as cluster centres called centroids
2. Assign each data point to the closest cluster by calculating its distance with respect to each centroid
3. Determine the new cluster centre by computing the average of the assigned points
4. Repeat steps 2 and 3 until none of the cluster assignments changes

From 1 above, if you set K = 1 then you select only 1 point and all your data points are close to this centroid so when you calculate the distance of the data points to the random K=1 point you chose, then when you calculate the average of the points it is the average of the whole data points and there is no moving from there.

In conclusion, never set K = 1.
